modules/widgets/src/splitter-widget.tsx

```diff
@@ -70,14 +79,62 @@ export class SplitterWidget extends WidgetImpl<SplitterWidgetProps> {
     render(
       <Splitter
         orientation={this.props.orientation}
-        initialSplit={this.props.initialSplit}
-        onChange={this.props.onChange}
+        initialSplitRatio={this.props.initialSplitRatio}
+        onChange={this.onSplitRatioChange}
         onDragStart={this.props.onDragStart}
         onDragEnd={this.props.onDragEnd}
       />,
       element
     );
   }
+
+  onSplitRatioChange = (splitRatio: number) => {
+    this.props.onSplitRatioChange(splitRatio);
+    this.splitRatio = splitRatio;
+    this._widgetManager!.viewsNeedUpdate = true;
+    this._widgetManager!.setNeedsRedraw('SplitterWidget');
+  };
+
+  filterViews(views: View[]): View[] {
+    const view1Index = views.findIndex(view => view.id === this.props.viewId1);
+    const view2Index = views.findIndex(view => view.id === this.props.viewId2);
+    if (view1Index !== -1 && view2Index !== -1) {
+      views = [...views];
+      const [viewProps1, viewProps2] = getViewPropsForSplitRatio(this.splitRatio);
+      views[view1Index] = cloneView(views[view1Index], viewProps1);
```

Comment:
Why not add `clone()` to the `View class, for consistency with `Layer`?

Reply:
> Why not add clone() to the View class, for consistency with Layer`?

Yes, that would be the "natural thing" to do. But for the purposes of keeping this PR small and attract as few tangential discussions as possible, I left that our for now. 

Added a TODO comment.